natlib/svcd.c

```diff
@@ -185,3 +187,84 @@ static int svcd_init( lua_State *L )
 
     return 0;
 }
+
+// Lua: storm.n.svcd_add_service ( svcd_id )
+// Add a new service to the service daemon
+// this must be called before SVCD.advertise()
+static int svcd_add_service( lua_State *L ) {
+
+    //TODO: Garbage collection?
+
+    lua_getglobal(L, "SVCD");
+
+    lua_pushstring(L, "blsmap");
+    lua_gettable(L, 2);
+    lua_pushvalue(L, 1); //svc_id @ index 4
+    lua_pushlightfunction(L, libstorm_bl_addservice);
+    lua_pushvalue(L, 1); //svc_id @ index 6
```

Comment:
I'm not sure what you're trying to do with the pushvalue() function. I see the comment says that you want to get svc_id that's at index 4 but you push the value that is at index 1. According to my understanding this should be pushing SVCD.blsmap onto the stack again since it is at index 1 of the stack.